pkg/virt-operator/install-strategy/strategy.go

```diff
@@ -280,13 +279,13 @@ func GenerateCurrentInstallStrategy(config *operatorutil.KubeVirtDeploymentConfi
 	strategy.services = append(strategy.services, components.NewPrometheusService(config.GetNamespace()))
 	strategy.services = append(strategy.services, components.NewApiServerService(config.GetNamespace()))
 	strategy.services = append(strategy.services, components.NewOperatorWebhookService(operatorNamespace))
-	apiDeployment, err := components.NewApiServerDeployment(config.GetNamespace(), config.GetImageRegistry(), config.GetImagePrefix(), config.GetApiVersion(), config.GetImagePullPolicy(), config.GetVerbosity(), config.GetExtraEnv())
+	apiDeployment, err := components.NewApiServerDeployment(config.GetNamespace(), config.GetImageRegistry(), config.GetImagePrefix(), config.GetApiVersion(), config.GetImagePullPolicy(), config.GetVerbosity(), config.GetExtraEnv() /*, config.GetPatches()*/)
 	if err != nil {
 		return nil, fmt.Errorf("error generating virt-apiserver deployment %v", err)
 	}
 	strategy.deployments = append(strategy.deployments, apiDeployment)
 
-	controller, err := components.NewControllerDeployment(config.GetNamespace(), config.GetImageRegistry(), config.GetImagePrefix(), config.GetControllerVersion(), config.GetLauncherVersion(), config.GetImagePullPolicy(), config.GetVerbosity(), config.GetExtraEnv())
+	controller, err := components.NewControllerDeployment(config.GetNamespace(), config.GetImageRegistry(), config.GetImagePrefix(), config.GetControllerVersion(), config.GetLauncherVersion(), config.GetImagePullPolicy(), config.GetVerbosity(), config.GetExtraEnv(), config.GetPatchesForResource("Deployment", components.ControllerDeploymentName))
```

Comment:
I wonder if instead of passing in the patches to each create function (like deployments, api, daemonset, etc...) if we could just loop over all the objects at the of this GenerateCurrentInstallStrategy instead.

Something like
```
// services
for i, entry := range strategy.services {
    strategy.services[i], err = applyPatches(entry, getPatchesForResource(.....))
    if err != nil {
        return nil, err
    }
}
// deployments
for i, entry := range strategy.deployments {
    strategy.deployments[i], err = applyPatches(entry, getPatchesForResource(.....))
    if err != nil {
        return nil, err
    }
}
```

maybe there's even a generic function that can be used to do this so it would be as simple as

```
genericApplyPatches(strategy.Services, config)
genericApplyPatches(strategy.Deployments, config)
.
.
.
```

I'm just trying to think if there's a way to consolidate all the patching logic to this generate function rather than spreading it around.

Reply:
Yeah I think that makes sense, I have updated apply patches to be more generic so it only needs to be called for each resource type that we want to allow patches on